src/routes/api.php

```diff
@@ -33,23 +33,31 @@
         Route::apiResource('users.presentations', 'API\PresentationController')
             ->only(['index', 'show', 'store'])
             ->shallow();
-        Route::apiResource('users.presentations', 'API\PresentationController')
-            ->only(['update', 'destroy'])
-            ->shallow()
-            ->middleware('presentation.roles:'.App\User\Role::PRESENTER());
+        Route::middleware(['presentation.roles:' . App\User\Role::PRESENTER()])->group(function () {
+            Route::apiResource('users.presentations', 'API\PresentationController')->only(['update', 'destroy'])->shallow();
+            Route::apiResource('presentations.polls', 'API\PollController')->only(['store'])->shallow();
+        });
+        Route::apiResource('presentations.polls', 'API\PollController')->only(['index', 'show'])->shallow();
         // presentation subscription management
         Route::middleware(['subscription.permission'])->group(function () {
             Route::post('presentations/{presentation}/users/{user}', 'API\PresentationUserController@subscribe')->name('presentations.users.subscribe');
             Route::delete('presentations/{presentation}/users/{user}', 'API\PresentationUserController@unsubscribe')->name('presentations.users.unsubscribe');
         });
         // manage user rights on presentations
         Route::put('presentations/{presentation}/users/{user}', 'API\PresentationUserController@changeRole')->name('presentations.change_role');
+
         // poll management
-        Route::apiResource('presentations.polls', 'API\PollController')->shallow();
-        Route::put('polls/{poll}/publish', 'API\PollController@publish')->name('polls.publish');
-        Route::get('polls/{poll}/results', 'API\PollController@results')->name('polls.results');
+        Route::middleware(['poll.roles:' . App\User\Role::PRESENTER()])->group(function () {
+            Route::put('polls/{poll}', 'API\PollController@update')->name('polls.update');
```

Comment:
it's better to use apiResource wrapper for "normal" routes like that instead of using php verbs by and

Reply:
See 0857d26